wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug Report/LogFilesProvider.swift

```diff
@@ -0,0 +1,109 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireCommonComponents
+import WireSystem
+import ZipArchive
+
+// sourcery: AutoMockable
+protocol LogFilesProviding {
+
+    /// Generates a zip file containing all log files and returns its data before removing the files
+    ///
+    /// - Returns: the log files archive data
+
+    func generateLogFilesData() throws -> Data
+
+    /// Generates a zip file containing all log files
+    /// 
+    /// - Returns: the log files archive URL
+
+    func generateLogFilesZip() throws -> URL
+
+    /// Clears the logs directory.
+    /// Call once you are done using the URL returned by `generateLogFilesZip` to clean up.
+
+    func clearLogsDirectory() throws
+
+}
+
+struct LogFilesProvider: LogFilesProviding {
+
+    // MARK: - Types
+
+    enum Error: Swift.Error {
+        case noLogs(description: String)
+    }
+
+    // MARK: - Properties
+
+    private let temporaryDirectory = NSTemporaryDirectory()
```

Comment:
Following on from @netbe message [here](https://github.com/wireapp/wire-ios/pull/1678#discussion_r1670712841) in my PR, I didn't realize that `NSTemporaryDirectory()` is not unique per call.. i.e calling this multiple times will return the same path. I think `temporaryDirectory` should be unique to `LogsFilesProvider` so I suggest appending  a unique path (e.g. `UUID().uuidString`) to `NSTemporaryDirectory()`.

Maybe this can all be done in `logsDirectory` and you could get rid of this property. Something like:

```swift
private var logsDirectory: URL = {
    let baseURL = URL(
        fileURLWithPath: NSTemporaryDirectory(),
        isDirectory: true
    )
    return baseURL.appendingPathComponent(UUID().uuidString).appendingPathComponent("logs")
}()
```

Reply:
we already do something similar but instead of having a path such as `tmp/<uuid>/logs/logs.zip` we have a path like `tmp/logs/<uuid>/logs.zip`. in `generateLogFilesZip()` we create an extra directory at the path of `logsDirectory`.

Reply:
My concern is that the API is a bit asymmetrical in that `generateLogFilesZip()` creates a log file zip but there is nothing to remove that log file zip... Instead we only have a way to remove all the files of `<user's temp>/logs`. For me this makes things less encapsulated than they could be. It requires us to keep in mind that `<user's temp>/logs` is reserved for this subsystem and could cause conflicts with other sub systems. For example we might find that some 3rd party library also makes use of `<user's temp>/logs` and then we would find that these sub-systems conflict with each other's behavior. 

So my proposed solution would be to make the logs directory unique to any given `LogsFileProvider` instance. 

This would allow for example separate parts of the app to generate zips and clean up simultaneously without affecting each other. 

In practice things will probably be fine but I do think there is a risk of running into a bug at some time with the current implementation.